recording_external_revenue_account/models/recording_external_revenue.py

```diff
@@ -0,0 +1,163 @@
+# © 2020 - today Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import api, models, fields, _
+from odoo.addons.queue_job.job import job
+from odoo.exceptions import ValidationError
+
+
+class RecordingExternalRevenue(models.Model):
+
+    _inherit = "recording.external.revenue"
+
+    is_posted = fields.Boolean(readonly=True)
+
+    def schedule_generate_journal_entries(self, company):
+        revenues_to_post = self.search([("is_posted", "=", False)]).filtered(
+            lambda r: r.company_id == company
+        )
+        for revenues in revenues_to_post:
+            revenues.with_delay().generate_journal_entry()
+
+    @job
+    def generate_journal_entry(self):
+        with self.env.do_in_onchange():
+            move = self._make_new_journal_entry()
+
+        move_vals = dict(move._cache)
+        move_vals["line_ids"] = [(0, 0, line._cache) for line in move.line_ids]
+        return self.env["account.move"].create(move_vals)
+
+    def _make_new_journal_entry(self):
+        move = self.env["account.move"].new(
+            {"journal_id": self._map_journal().id, "date": self.operation_date,}
+        )
+
+        revenue_line = self._make_new_revenue_move_line()
+        move.line_ids = revenue_line
+
+        self._set_tax_base_amount(revenue_line)
+        self._add_tax_move_lines(move)
+        self._set_revenue_amount(revenue_line)
+
+        self._add_receivable_move_line(move)
+
+        return move
+
+    def _add_tax_move_lines(self, move):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://app.codacy.com/manual/numigi/odoo-recording-addons/pullRequest?prid=5641900)